artifacts/artifact.py

```diff
@@ -82,6 +82,11 @@ def AppendSource(self, type_indicator, attributes):
       raise errors.FormatError(
           u'Unsupported type indicator: {0}.'.format(type_indicator))
 
-    source_object = source_type_class(**attributes)
+    try:
+      source_object = source_type_class(**attributes)
```

Comment:
> raise errors.FormatError("Invalid artifact definition for %s: %s" % (self.name, e))

uses % instead of format which is used every else in the code base.